.github/workflows/style.yaml

```diff
@@ -19,13 +19,14 @@ jobs:
       uses: actions/setup-python@v2
       with:
         python-version: '3.10'
-    - name: Install pip dependencies
+    - name: Install pip dependencies via poetry
       run: |
-        pip install --pre 'rasterio>=1.0.16'
-        pip install .[style]
-        pip list
+        pip install poetry
+        poetry install
+        poetry show
     - name: Run black checks
-      run: black . --check --diff
+      run: poetry run black . --check --diff
```

Comment:
This seems very wasteful instaling the whole bucket of dependencies just to run `black`. Same with the other linters below. Is there a need to use pinned versions for the linter?

Could we just do `pip install black`, or consider something like [`super-linter`](https://github.com/github/super-linter/tree/v4.9.4)? Maybe something to do in a separate PR.

Reply:
Yes, for black/flake8/isort/pydocstyle/pyupgrade we can just install those deps (please open a PR), for mypy/pytest we do need all deps.